x-pack/platform/plugins/shared/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -1228,4 +1243,77 @@ export class AlertsClient {
       fields,
     };
   }
+
+  public getRuleTypeIds(ruleTypeIds: string | string[] | undefined): string[] {
+    // fetch all rule types if no specific rule type Ids are provided
+    if (!ruleTypeIds || ruleTypeIds.length === 0) {
+      const registeredRuleTypes = this.getRuleList();
+
+      if (!registeredRuleTypes) {
+        throw Boom.badRequest('No rule types are registered');
+      }
+
+      return Array.from(registeredRuleTypes.keys());
+    }
+
+    return Array.isArray(ruleTypeIds) ? ruleTypeIds : [ruleTypeIds];
+  }
+
+  public async getAlertFields(
+    ruleTypeIds: string | string[]
```

Comment:
Should we validate `ruleTypeIds` to see if they match any registered rule types?

Reply:
The api is internal so i think you are right 🤔 I guess the distinction between "this type does not exist" vs "you are not authorized to access this type" does not matter here.